### PR TITLE
Pebble doc and test updates

### PR DESCRIPTION
**Pull request by cjdcordeiro**
- [x] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?

-----

Fixes some formatting issues with the docs while also updating both the docs and the spread tests to be according to the most recent Pebble additions (like `--args`)

### CODE REVIEW

**Review by SamirPS — APPROVED** [review]

File: tests/spread/general/run-user/rockcraft.yaml [hunk 1]
```diff
@@ -8,7 +8,7 @@ run-user: _daemon_
 services:
   test:
     override: replace
-    command: /usr/bin/sleep 10
+    command: /usr/bin/sleep [ 1 ]
```
Comment (tigarmo) [inline tests/spread/general/run-user/rockcraft.yaml:11]: ah nice

**Review by tigarmo — APPROVED** [review]

File: tests/spread/general/run-user/rockcraft.yaml (hunk 1 above)
Comment (cjdcordeiro) [reply tests/spread/general/run-user/rockcraft.yaml:11]: ;)

File: tests/spread/general/craftctl/task.yaml [hunk 2]
```diff
@@ -15,5 +15,5 @@ execute: |
   rm craftctl-test_latest_craftctl.rock
   docker images craftctl-test:latest
   docker run --rm --entrypoint /usr/bin/hello craftctl-test:latest
-  docker run --rm --entrypoint pebble craftctl-test:latest help
-  docker run --rm --entrypoint pebble craftctl-test:latest --help
\ No newline at end of file
+  docker run --rm craftctl-test:latest help
+  docker run --rm craftctl-test:latest --help
```
Comment (lengau) [inline tests/spread/general/craftctl/task.yaml:19]:
```suggestion
  docker run --rm craftctl-test:latest --help
  
```

(Again, just github complaining)

**Review by lengau — APPROVED** [review]
Looks great! Please let me know your thoughts on the comment I made, but I'm happy for you to merge as-is (I just wanted to point it out since I'm sure somebody else will read these docs before they've had their coffee too.)

File: docs/how-to/code/convert-to-pebble-layer/task.yaml [hunk 3]
```diff
@@ -19,7 +19,7 @@ execute: |
   rm custom-nginx-rock_latest_amd64.rock
 
   # [docs:docker-run]
-  docker run -d --name nginx-pebble-service -p 8080:80 custom-nginx-rock:latest start nginx
+  docker run -d --name nginx-pebble-service -p 8080:80 custom-nginx-rock:latest --args nginx -g 'daemon off;' \; start nginx
```
[the remark this answers is not in the document]
Comment (cjdcordeiro) [reply docs/how-to/code/convert-to-pebble-layer/task.yaml:22]: This is Pebble's syntax for `--args`. It is inspired by `find`'s exec commands and it was the only way we found to be able to start a ROCK while giving arguments to services and issuing client commands in one go.

[the remark this answers is not in the document]
Comment (cjdcordeiro) [reply docs/how-to/code/convert-to-pebble-layer/task.yaml:22]: I'm closing this to allow it to be merged. But pls open a new issue if you feel this syntax is too alien and should somehow be documented in Rockcraft in addition to the existing Pebble docs.

[the remark this answers is not in the document]
Comment (lengau) [reply docs/how-to/code/convert-to-pebble-layer/task.yaml:22]: That makes sense, thanks for explaining! The pebble syntax makes sense.

At most I think we could do a change to `convert-to-pebble-layer.rst` to mention it just above, but I read the doc page this morning in context and I think the context from the page is enough.

**Pull request merged**